File: ruby/sqlite3_sandbox.rb
#!/usr/bin/env ruby
# SQLite3 to CSV in Ruby.
# 
# If that's all you want, you can also use:
# 
#     .mode csv
# 
# when using SQLite3.
# 
# Author: Benjamin Oakes <hello@benjaminoakes.com>

require 'csv'
require 'stringio'

require 'rubygems'
require 'sqlite3'

def select_all_from(table_name)
  database = SQLite3::Database.new(File.expand_path('~/Outbox/output_notest.sqlite3'))

  rows = []

  database.execute('select * from ' + table_name) do |row|
    rows << row
  end
  
  return rows
end

csv_io = StringIO.new('')

CSV::Writer.generate(csv_io) do |csv|
  select_all_from('my_table').each do |row|
    csv << row
  end
end

csv_io.rewind
puts csv_io.readlines

File: perl/sqlite3_sandbox.pl
# SQLite3 to CSV in Perl.
# 
# If that's all you want, you can also use:
# 
#     .mode csv
# 
# when using SQLite3.
# 
# Author: Benjamin Oakes <hello@benjaminoakes.com>
use Data::Dumper;
use DBI;

my $database_path = 'output.sqlite3';

$database = DBI->connect(
  "dbi:SQLite:dbname=$database_path",
  "",
  "",
  {
    RaiseError => 1,
    AutoCommit => 0,
  }
) or die "Couldn't connect to database: " . DBI->errstr;

my $result = $database->prepare('select * from my_view;')
    or die "Couldn't prepare query: " . $database->errstr;

$result->execute
    or die "Couldn't execute query: " . $result->errstr;

my $fields = join(',', @{ $result->{NAME_lc} });
print "$fields\n";

while (my $row = $result->fetchrow_arrayref) {
    my $csv = join(',', @$row); # NOTE Please use a real CSV solution in a production app.
    print "$csv\n";
}

$result->finish;

$database->disconnect;

